jobrunner/executors/local.py

```diff
@@ -256,7 +256,15 @@ def finalize_job(job):
         message = config.DATABASE_EXIT_CODES.get(exit_code)
     # No database error, check for other known exit codes
     if message is None:
```

Comment:
This is a really nice UX improvement. I think the `OOMKilled` check could be done before the `allow_database_access` check though, because db jobs can in principle get OOM-killed to. I also wonder if we shouldn't merge the DATABASE_EXIT_CODES and EXIT_CODES dicts when looking up the error messages for database jobs, so that database jobs still get the standard messages for standard error codes. For instance, I can imagine an admin killing a database job and it would be nice to give the right message there.

Reply:
That's a good idea. I'll do a follow up PR

Reply:
#414